topology/src/main/java/io/camunda/zeebe/topology/changes/PartitionLeaveApplier.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.topology.changes;
+
+import io.atomix.cluster.MemberId;
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import io.camunda.zeebe.scheduler.future.CompletableActorFuture;
+import io.camunda.zeebe.topology.changes.TopologyChangeAppliers.OperationApplier;
+import io.camunda.zeebe.topology.state.ClusterTopology;
+import io.camunda.zeebe.topology.state.MemberState;
+import io.camunda.zeebe.topology.state.PartitionState;
+import io.camunda.zeebe.util.Either;
+import java.util.Map;
+import java.util.function.UnaryOperator;
+
+/**
+ * A partition leave operation is executed when a member wants to stop replicating a partition. This
+ * is allowed only when the member is already replicating the partition.
+ */
+record PartitionLeaveApplier(
+    int partitionId,
+    MemberId localMemberId,
+    PartitionTopologyChangeExecutor partitionTopologyChangeExecutor)
+    implements OperationApplier {
+
+  @Override
+  public Either<Exception, UnaryOperator<MemberState>> init(
```

Comment:
:thought_balloon: I think we briefly talked about a validation such as "partition should have at least one remaining member" but not sure if we want to do that now or just leave it for now.

Reply:
Right, we can add it later. Also, would raft reject this?

Reply:
I don't think so